include/up-cpp/utils/Expected.h

```diff
@@ -22,87 +23,161 @@ namespace uprotocol::utils {
+	constexpr explicit Unexpected(E&& rhs) : storage(std::move(rhs)) {}
+
+	template <class... Args>
+	constexpr explicit Unexpected(std::in_place_t, Args&&... args)
+	    : storage(args...) {}
+
+	// template <class U, class... Args>
+	// constexpr explicit Unexpected(std::in_place_t, std::initializer_list<U>
+	// il, Args&&... args)
+
+	constexpr const E& error() const& noexcept { return std::move(storage); }
+
+	// this one is checked
+	constexpr E& error() & noexcept { return storage; }
+
+	constexpr const E&& error() const&& noexcept { return std::move(storage); }
+
+	constexpr E&& error() && noexcept { return std::move(storage); }
+};
 
 /// @brief A stripped-down version of std::expected from C++23.
-template <typename T, typename E>
+template <typename E, typename U>
 struct Expected {
-	using value_type = T;
-	using error_type = E;
-
-	constexpr Expected();
-	constexpr Expected(const Expected& other);
-	constexpr Expected(Expected&& other) noexcept;
-
-	constexpr explicit Expected(T&& v);
-	constexpr explicit Expected(E&& e);
-
-	Expected& operator=(const Expected& other);
-	Expected& operator=(Expected&& other);
-
-	Expected& operator=(const T& other);
-	Expected& operator=(T&& other);
-
-	Expected& operator=(const E& other);
-	Expected& operator=(E&& other);
-
-	constexpr explicit operator bool() const noexcept;
-	constexpr bool has_value() const noexcept;
-
-	constexpr T& value() &;
-	constexpr const T& value() const&;
-	constexpr T&& value() &&;
-	constexpr const T&& value() const&&;
-
-	constexpr const E& error() const& noexcept;
-	constexpr E& error() & noexcept;
-	constexpr const E&& error() const&& noexcept;
-	constexpr E&& error() && noexcept;
-
-	constexpr T value_or(T&& default_value) const&;
-	constexpr T value_or(T&& default_value) &&;
-
-	template <typename F>
-	constexpr auto and_then(F&& f) &;
-	template <typename F>
-	constexpr auto and_then(F&& f) const&;
-	template <typename F>
-	constexpr auto and_then(F&& f) &&;
-	template <typename F>
-	constexpr auto and_then(F&& f) const&&;
-
-	template <typename F>
-	constexpr auto or_else(F&& f) &;
-	template <typename F>
-	constexpr auto or_else(F&& f) const&;
-	template <typename F>
-	constexpr auto or_else(F&& f) &&;
-	template <typename F>
-	constexpr auto or_else(F&& f) const&&;
+	std::variant<E, Unexpected<U>> storage;
+
+public:
+	constexpr Expected() { storage = E(); }
+	constexpr Expected(E&& rhs) : storage(std::move(rhs)) {}
+	constexpr Expected(Unexpected<U>&& rhs) : storage(std::move(rhs)) {}
+
+	constexpr Expected(const Unexpected<U>& rhs) {
+		std::get<Unexpected<U>>(storage).storage = std::move(rhs);
+	}
+
+	template <class X = E>
+	constexpr explicit Expected(X&& rhs) {
+		std::get<E>(storage) = std::forward<X>(rhs);
+	}
+
+	template <class X = U>
+	constexpr explicit Expected(Unexpected<X>&& rhs) {
+		std::get<E>(storage) = rhs.error();
+	}
+
+	constexpr Expected(const Expected& rhs) { storage = rhs.storage; }
```

Comment:
If [this change](https://github.com/eclipse-uprotocol/up-cpp/pull/124#discussion_r1628657040) is applied, then this comment will be obsolete. Otherwise the problem still exists.